features/bootstrap/FeatureContext.php

```diff
@@ -183,26 +185,35 @@ public function iSetEnvironmentVariable(PyStringNode $value)
     public function iRunBehat($argumentsString = '')
     {
         $argumentsString = strtr($argumentsString, array('\'' => '"'));
-
-        $this->process->setWorkingDirectory($this->workingDir);
-        $this->process->setCommandLine(
-            sprintf(
-                '%s %s %s %s',
-                $this->phpBin,
-                escapeshellarg(BEHAT_BIN_PATH),
-                $argumentsString,
-                strtr($this->options, array('\'' => '"', '"' => '\"'))
-            )
+        $cmd =  sprintf(
+            '%s %s %s %s',
+            $this->phpBin,
+            escapeshellarg(BEHAT_BIN_PATH),
+            $argumentsString,
+            strtr($this->options, array('\'' => '"', '"' => '\"'))
         );
-
+        if (method_exists('\\Symfony\\Component\\Process\\Process','fromShellCommandline')) {
+            $process = Process::fromShellCommandline($cmd);
+        } else {
+            $process = new Process(null);
+            $process->setCommandLine($cmd);
+        }
+        $process->setTimeout(20);
+    
+        $process->setWorkingDirectory($this->workingDir);
+    
+        $env = $this->processEnv;
         // Don't reset the LANG variable on HHVM, because it breaks HHVM itself
         if (!defined('HHVM_VERSION')) {
-            $env = $this->process->getEnv();
             $env['LANG'] = 'en'; // Ensures that the default language is en, whatever the OS locale is.
-            $this->process->setEnv($env);
         }
-
-        $this->process->run();
+        if($this->input !== null){
+            $process->setInput($this->input);
+        }
+        
+        $process->setEnv($env);
+        $this->process = $process;
+        $process->run();
```

Comment:
Shouldn't this be `$this->process->run()`? Or `$process->run()` can be before we assign the variable `$this->process`?